mono/io-layer/wait.c

```diff
@@ -244,21 +234,7 @@ guint32 SignalObjectAndWait(gpointer signal_handle, gpointer wait,
 	guint32 ret = 0, waited;
 	int thr_ret;
 	gboolean apc_pending = FALSE;
-	gpointer current_thread = mono_thread_info_get_handle (mono_thread_info_current ());
 	gint64 now, end;
-	
-	if (current_thread == NULL) {
-		SetLastError (ERROR_INVALID_HANDLE);
-		return(WAIT_FAILED);
-	}
-
-	if (signal_handle == _WAPI_THREAD_CURRENT) {
```

Comment:
We do not create handle with value _WAPI_THREAD_CURRENT any more. So no handle can have this value.